hudi-common/src/main/java/org/apache/hudi/avro/AvroSchemaUtils.java

```diff
@@ -76,6 +101,19 @@ public static Schema resolveUnionSchema(Schema schema, String fieldSchemaFullNam
     return nonNullType;
   }
 
+  /**
+   * Returns true in case provided {@link Schema} is nullable (ie accepting null values),
+   * returns false otherwise
+   */
+  public static boolean isNullable(Schema schema) {
+    if (schema.getType() != Schema.Type.UNION) {
+      return false;
+    }
+
+    List<Schema> innerTypes = schema.getTypes();
+    return innerTypes.size() > 1 && innerTypes.stream().anyMatch(it -> it.getType() == Schema.Type.NULL);
+  }
+
```

Comment:
HiveAvroSerializer  also has isNullableType,
maybe it is better to keep only one

Reply:
Good call, would prefer to take this as a follow-up to not overload this PR even further

Reply:
can you file a follow up jira and track all such deferred items. I also saw java docs in some of them are missing.

Reply:
Forgot to push commit updating TODOs